src/backend/marsha/core/api/video.py

```diff
@@ -544,6 +546,156 @@ def pairing_secret(
         serializer = serializers.LivePairingSerializer(instance=live_pairing)
         return Response(serializer.data)
 
+    @action(
+        methods=["patch"],
+        detail=True,
+        url_path="start-sharing",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject
+            & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+        ],
+    )
+    # pylint: disable=unused-argument
+    def start_sharing(
+        self,
+        request,
+        pk=None,
+    ):
+        """Starts a media live sharing in a live stream.
+
+        Broadcasts a xmpp message to all users in the video room.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the serialized video.
+        """
+        video = self.get_object()
+        if video.active_shared_live_media_id is not None:
+            return Response({"detail": "Video is already sharing."}, status=400)
+
+        sharedlivemedia = SharedLiveMedia.objects.get(
+            id=request.data.get("sharedlivemedia")
+        )
+        if sharedlivemedia.upload_state != defaults.READY:
+            return Response({"detail": "Shared live media is not ready."}, status=400)
+
+        if sharedlivemedia.video != video:
+            return Response(
+                {"detail": "Shared live media is not related to this video."},
+                status=400,
+            )
+
+        video.active_shared_live_media = sharedlivemedia
+        video.active_shared_live_media_page = 1
+        video.save()
+        serializer = self.get_serializer(video)
+
+        broadcast_message(str(video.id), json.dumps(serializer.data))
+        return Response(serializer.data)
+
+    @action(
+        methods=["patch"],
+        detail=True,
+        url_path="navigate-sharing",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject
+            & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+        ],
+    )
+    # pylint: disable=unused-argument
+    def navigate_sharing(
+        self,
+        request,
+        pk=None,
+    ):
+        """Changes page of the media live shared in a live stream.
+
+        Broadcasts a xmpp message to all users in the video room.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the serialized video.
+        """
+        video = self.get_object()
+
+        if not video.shared_live_medias.exists():
```

Comment:
I don't understand why you check that ?
You should check if there is an `active_shared_live_media` instead, don't you think ? You can't change a page if there is not active shared live media